packages/graphql-yoga/src/plugins/request-validation/use-check-graphql-query-params.ts

```diff
@@ -133,11 +149,78 @@ export function isValidGraphQLParams(params: unknown): params is GraphQLParams {
   }
 }
 
+export function checkOperationName(
+  operationName: string | undefined,
+  document: DocumentNode,
+): void {
+  const operations = listOperations(document);
+
+  if (operationName != null) {
+    for (const operation of operations) {
+      if (operation.name?.value === operationName) {
+        return;
+      }
+    }
+
+    throw createGraphQLError(
+      `Could not determine what operation to execute. There is no operation "${operationName}" in the query.`,
+      {
+        extensions: {
+          http: {
+            spec: true,
+            status: 400,
+          },
+        },
+      },
+    );
+  }
+
+  operations.next();
+  // If there is no operation name, we should have only one operation
+  if (!operations.next().done) {
+    throw createGraphQLError(
+      `Could not determine what operation to execute. The query contains multiple operation, an operation name should be provided.`,
+      {
+        extensions: {
+          http: {
+            spec: true,
+            status: 400,
+          },
+        },
+      },
+    );
+  }
+}
+
+export function isValidOperationName(
+  operationName: string | undefined,
+  document: DocumentNode,
+): boolean {
+  try {
+    checkOperationName(operationName, document);
+    return true;
+  } catch {
+    return false;
+  }
+}
```

Comment:
Not sure it's needed, just wanted to implement the same pattern that was used for the check of the params

Reply:
Where is this function used?